source/grid/grid_in.cc

```diff
@@ -2009,23 +2009,62 @@ GridIn<dim, spacedim>::read_msh(std::istream &in)
 
   unsigned int n_vertices;
   unsigned int n_cells;
+  int          n_entity_blocks = 1;
```

Comment:
```suggestion
  unsigned int          n_entity_blocks = 1;
```
This could be an unsigned int, no reason for this to have a negative value.